internal/signerverifier/ssh/ssh.go

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: Apache-2.0
+package ssh
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"encoding/base64"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/hiddeco/sshsig"
+	"golang.org/x/crypto/ssh"
+)
+
+// SSH Key/Verifier
+type Key struct {
+	KeyType string
+	KeyVal  KeyVal
+	Scheme  string
+	keyID   string
+}
+
+type KeyVal struct {
+	Public string
+}
+
+// SSH Key.Verify() implementation of dsse.Verifier interface.
+func (k *Key) Verify(_ context.Context, data []byte, sig []byte) error {
+	pub, err := parseSSH2Body(k.KeyVal.Public)
+	if err != nil {
+		return fmt.Errorf("failed to parse ssh public key material: %w", err)
+	}
+
+	signature, err := sshsig.Unarmor(sig)
+	if err != nil {
+		return fmt.Errorf("failed to parse ssh signature: %w", err)
+	}
+
+	message := bytes.NewReader(data)
+
+	// ssh-keygen uses sha512 to sign with **any*** key
+	hash := sshsig.HashSHA512
+	if err = sshsig.Verify(message, signature, pub, hash, "gittuf"); err != nil {
```

Comment:
hm, this is a problem; we'd hit this to verify a commit or tag signature as well, when the namespace is not in our control. I wonder if the namespace needs to become part of the key and we have to `SetNamespace()` wherever we're using this? It's not super clean; we'd have to special case the SSH verifier, but I don't see another solution atm.

Reply:
We could also piggyback on git's namespace, i.e. `git`:
https://github.com/gittuf/gittuf/pull/414

Reply:
> i.e., just use git even for attestations? That could certainly work

Yes!

> might be unexpected for any other verifier

I think, we can and should hide the namespace from the verifier. It's a spill over from a different trust model: 
https://github.com/openssh/openssh-portable/blob/8054b906983ceaed01fabd8188d3dac24c05ba39/PROTOCOL.sshsig#L53-L57

AFAICS we don't need to consider a namespace value to determine the eligibility of a signature.